test/test_nestedtensor.py

```diff
@@ -6166,6 +6166,98 @@ def test_unbind_backward(self, device, dtype):
+        self.assertEqual(masked_nt.lengths(), None)
+
+        njt = nested_view_from_values_offsets(
+            values=torch.arange(6, device=device), offsets=offsets
+        )
+        mask = torch.tensor([True, False, True], device=device)
+
+        masked_nt = njt.masked_select(mask)
+        self.assertEqual(njt.values(), torch.arange(6, device=device))
+        self.assertEqual(masked_nt.values(), torch.tensor([0, 1, 5], device=device))
+        self.assertEqual(
+            masked_nt.offsets(),
+            torch.tensor([0, 2, 2, 3]),
+        )
+        self.assertEqual(masked_nt.lengths(), None)
+
+    @skipIfTorchDynamo(
+        "Fails on returning a fresh tensor but this is expected behavior"
+    )
+    def test_nested_masked_select(self, device):
+        t = torch.randn([3, 3], device=device)
+        mask = torch.tensor([False], device=device)
+
+        njt = torch.nested.masked_select(t, mask)
+        self.assertEqual(njt.values(), torch.tensor([], device=device))
+        self.assertEqual(njt.lengths(), torch.tensor([0, 0, 0], device=device))
+        self.assertEqual(njt.offsets(), torch.tensor([0, 0, 0, 0], device=device))
+
+        mask = torch.tensor([[False], [False], [True]])
+        njt = torch.nested.masked_select(t, mask)
+        self.assertEqual(njt.values(), t[-1], atol=0.1, rtol=0.1)
+        self.assertEqual(njt.lengths(), torch.tensor([0, 0, 3], device=device))
+        self.assertEqual(njt.offsets(), torch.tensor([0, 0, 0, 3], device=device))
+
+        mask = torch.tensor(
+            [[False, False, True], [True, False, True], [False, False, True]],
+            device=device,
+        )
+        njt = torch.nested.masked_select(t, mask)
+        self.assertEqual(njt.values(), t.masked_select(mask))
+        self.assertEqual(njt.lengths(), torch.tensor([1, 2, 1], device=device))
+        self.assertEqual(njt.offsets(), torch.tensor([0, 1, 3, 4], device=device))
+
+        t = torch.randn([2, 3, 3, 1], device=device)
+        mask = torch.tensor(
+            [
+                [
+                    [[True], [False], [True]],
+                    [[True], [False], [True]],
+                    [[True], [False], [True]],
+                ],
+                [
+                    [[False], [True], [True]],
+                    [[False], [True], [True]],
+                    [[True], [True], [True]],
+                ],
+            ],
+            device=device,
+        )
+        njt = torch.nested.masked_select(t, mask)
+        self.assertEqual(njt.values(), t.masked_select(mask))
+        self.assertEqual(
+            njt.lengths(),
+            torch.tensor(
+                [1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1], device=device
+            ),
+        )
+        self.assertEqual(
+            njt.offsets(),
+            torch.tensor(
+                [0, 1, 1, 2, 3, 3, 4, 5, 5, 6, 6, 7, 8, 8, 9, 10, 11, 12, 13],
+                device=device,
+            ),
+        )
+
```

Comment:
Btw the systematic op-info based testing for NJTs has landed. We should lean on that if possible (see `torch/testing/_internal/opinfo/definitions/nested.py`) and only have one-off tests if what we want to test cannot be covered by the systematic ones. 

If you add the sample inputs for your ops there, it would also run the forward/backward/forward+compile/backward+compile tests for your op upon running `test/test_nestedtensor.py -k TestNestedTensorOpInfo`.

Reply:
I'll rework the test

Reply:
So I got the cases updated here, but two things:

1) These don't actually verify outputs for the operators (unless I'm missing something.)
2) Is there a way to enable dynamic shapes? (for now I put the test cases as expected failures but verified that they work in eager mode.)

I'll keep the older test versions for now.

Reply:
> These don't actually verify outputs for the operators 

The `test_forward` test compares the forward of njt with output with an unbind-based fallback, i.e., unbind on batch dim and compute output row by row.

> Is there a way to enable dynamic shapes? (for now I put the test cases as expected failures but verified that they work in eager mode.)

Easiest way for now is to update our compile tests to always `torch._dynamo.config.capture_dynamic_output_shape_ops = True` e.g. with `torch._dynamo.config.patch(` context/decorator.
Even the future if we want to document/test specifically which ops are dynamic output shape, we can update the opinfo entry to hold extra metadata.